tests/functional_tests/iterator_tests/test_trivial.py

```diff
@@ -167,3 +167,30 @@ def test_vertical_shift_unstructured(program_processor):
 
     if validate:
         assert np.allclose(inp_s[:, 1:], np.asarray(out_s)[:, :-1])
+
+
+def test_unapplied_op():
```

Comment:
Can you parameterize this test for `[plus, minus, divides, ...]`? Otherwise we can't really know if you used the correct C++ function objects.

Reply:
done. removed this test again and moved to test_builtins